src/adk/api/local_api.py

```diff
@@ -426,8 +426,10 @@ def is_application_valid(self, application_name: str) -> ErrorDictType:
             Returns empty list when all validations passes
             Returns dict containing error messages of the validations that failed
         """
+
```

Comment:
How it is solved now is not maintainable. Now the validation steps are scattered a bit for different commands. What I propose is for example for experiments:
1. In command_list for all relevant experiment commands (delete, validate, run, results) a method is called. path and experiment_name goes in and returns a valid experiment_path and a valid experiment_name. In this method we do: 

if experiment_name is not None:
....validate_path_name("Experiment", experiment_name)
experiment_path = path / experiment_name if experiment_name is not None else path
if not experiment_path.is_dir():
....raise ExperimentDirectoryNotValid(str(experiment_path))
return experiment_path, experiment_name

2. Depending on what is needed we call processor with one or both values.

For application (validate, delete) the same can be done but there we need a valid application_path and a valid application_name as a result